docs/automated_testing.md

```diff
@@ -0,0 +1,509 @@
+
+If we see that people are finding this framework useful, we can improve on that.
+
+
+### The test failed on or after ‘upload error artifacts’ 
+{#the-test-failed-on-or-after-‘upload-error-artifacts’}
+
+Something probably went wrong with Github. Or maybe they have a maximum amount that they can download that we haven’t yet dug up in their documentation. Try rerunning the tests and, if it fails the same way a second time, get in touch with us.
+
+
+### The error says it “timed out“ 
+{#the-error-says-it-“timed-out“}
+
+That’s a stock system error. Some **Step** took too long to finish in a way for which we have not yet created a custom error message or cannot detect. It is often a page that took too long to load. It sometimes happens when the Project got deleted in the middle of the test, or when the [`Run npm run setup` phase doesn’t work correctly](#bookmark=id.y1ibk27fo4jy).
+
+
+### Some other mysterious error 
+{#some-other-mysterious-error}
+
+
+
+1. You can rerun the test
+2. Try to search for the text of the error online (don’t spend more than 20 min on this, though)
+3. Ask one of us. Remember that this framework is under development. Something might be wrong with our code.
+
+-->
+
+<!--
+
+### Translations
+
+Always give your `choices` values that are separate from their labels and will never be translated.
+
+In docassemble, you can write `choices` a few different ways.
+
+Bad
+
+```
+question: |
+  What is your favorite color?
+fields:
+  - Colors: favorite_color
+    input type: radio
+    choices:
+      - green
+      - red
+      - blue
+```
+
+Good
+
+```
+question: |
+  What is your favorite color?
+fields:
+  - Colors: favorite_color
+    input type: radio
+    choices:
+      - green: green
+      - red: red
+      - blue: blue
+```
+
+If your field is like the Bad example, the label of a choice is also its value and the translations will break the tests. This is because the label will be translated. Since the label written this way is also the value, the value in the translated version of the interview will be different than the value you wrote in for your test.
+
+If your field is like the Good example, the value of the choice should never get translated so it should always be the same and the tests will be happy.
+
+-->
+
+
+## Test instructions details
+
+### Add a new test
+
+Go to your Playground > the dropdown Folders menu > Sources.
+
+Add a new file that ends in the extension `.feature`. Ex: `has_children.feature`
+
+The next time you commit, your test will be run. You can add and edit multiple test files.
```

Comment:
Is setting up the github action part of the first section, "Setup", that's commented out?

Reply:
Yes. In fact, the developer doesn't have to actively set up any actions. They just go through that setup interview. It should be self-explanatory when they run it, but I haven't user tested it yet. That said, the current PR is an MVP for making something to help conversations between the folks in the AL project about writing tests. I can do setup for our repos for now.

Let me know if we should discuss more. If this is a satisfactory answer, feel free to resolve this.

Reply:
That's fine, skimmed right over that sentence, and it explains the part I thought was missing. Thanks!